src/parser/smt2/Smt2.g

```diff
@@ -581,7 +580,8 @@ sygusCommand returns [std::unique_ptr<CVC4::Command> cmd]
       sygusGrammar[grammar, synthFunFactory->getSygusVars(), fun]
     )?
     {
-      cmd = synthFunFactory->mkCommand(grammar);
+      cmd = synthFunFactory->mkCommand(grammar.get());
+      PARSER_STATE->getAllocGrammars().push_back(std::move(grammar));
```

Comment:
This isn't necessary if we hide the allocation details of Grammar.

Reply:
Removed!